console/src/components/Common/utils/v1QueryUtils.ts

```diff
@@ -98,6 +98,62 @@ export const getDropPermissionQuery = (
   };
 };
 
+export const getInsertUpQuery = (
+  tableDef: TableDefinition,
+  insertion: object
+) => {
+  return {
+    type: 'insert',
+    args: {
+      table: tableDef,
+      returning: [],
+      object: insertion,
+    },
+  };
+};
+
+type DeleteObjectWithId = {
+  id: number;
+};
+
+type PrimaryKeyInfo = {
+  table_name: string;
+  table_schema: string;
+  constraint_name: string;
+  columns: Array<string>;
+};
+
+export const getInsertDownQuery = (
+  tableDef: TableDefinition,
+  insertion: unknown,
+  primaryKeyInfo: PrimaryKeyInfo
+) => {
+  let whereClause = {};
+  const isIDPresent = (insertion as DeleteObjectWithId)?.id;
+
+  if (!isIDPresent) {
+    const primaryKeys = primaryKeyInfo.columns;
+    if (primaryKeys.length) {
+      const chosenKey = primaryKeys[0];
+      whereClause = { [chosenKey]: (insertion as any)[chosenKey] };
+    } else {
+      // this should never the case since a primary key is
+      // required for creating a table
+      return null;
+    }
```

Comment:
@beerose this is extreme "safe" code. I think we should remove. What say?

Reply:
This can happen because you can create a table without any primary keys with SQL. In this case, we should build `whereClause` based on each column. Check [this](https://github.com/hasura/graphql-engine/blob/master/console/src/components/Services/Data/TableBrowseRows/ViewRows.js#L226).
We should use similar logic here, which also means that we should get rid of `isIDPresent`. That's because we can't assume that most tables have `id` column. What's more, the table can have `id` column, but it may not be a primary key. In other words, we need to stay away from any assumptions regarding a table definition.